server/src/test/java/org/elasticsearch/index/mapper/MapperServiceTests.java

```diff
@@ -1707,6 +1707,78 @@ public void testExpandDottedNotationToObjectMappers() throws IOException {
             }""");
     }
 
+    public void testMergeDottedAndNestedNotation() throws IOException {
+        CompressedXContent mapping1 = new CompressedXContent("""
+            {
+              "properties": {
+                "parent.child": {
```

Comment:
To be safe, could you also add a test for the invers? Have the nested notation first (mapping2) and then the dot notation (mapping1). In theory, should not make a difference but I rather have one test to much then too little.

Reply:
Sure! I added that scenario with [cb28cdb](https://github.com/elastic/elasticsearch/pull/106077/commits/cb28cdbe5a47b0d3b527087acc192abfa387d266).